lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -682,50 +682,46 @@ namespace Js
         // NOTE Don't set the throwing exception here, because we might need to box it and will cause a nested stack walker
         // instead, return it to be set in WalkStackForExceptionContext
 
-        if (stackCrawlLimit == 0)
+        JavascriptExceptionContext::StackTrace *stackTrace = RecyclerNew(scriptContext.GetRecycler(), JavascriptExceptionContext::StackTrace, scriptContext.GetRecycler());
+        if (stackCrawlLimit > 0)
```

Comment:
You moved RecyclerNew(...StackTrace...) outside of BEGIN/END_TRANSLATE_EXCEPTION block. This changed OOM logic. This code is called during exception throw. If it OOMs here and throws a different exception, that would be undesirable.